### PR TITLE
Add default branch flag

### PR DESCRIPTION
**Pull request by andreapiras**
- Add defaultBranch flag with `default = master`
- Allow users to specify a different default branch other than master branch

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,19 +33,22 @@ deploy-to-github-pages [...options]
 ```javascript
 const deploy = require('deploy-to-github-pages');
 
-deploy(options).catch(err => { console.log(err); })
+deploy(options).catch(err => {
```
Comment (oliverviljamaa) [inline README.md:36]: :cool:

File: lib/deployment/preparation/preparation.js [hunk 2]
```diff
@@ -7,28 +7,34 @@ shell.set('-e');
...
-  if (!isMaster(branch)) {
+function createBranchDirectoryIfNeeded(deployDirectory, branch, defaultBranch) {
+  if (!isDefaultBranch(branch, defaultBranch)) {
     shell.mkdir('-p', getBranchDirectory(deployDirectory));
   }
 }
 
-function isMaster(branch) {
-  return branch === 'master';
+function isDefaultBranch(branch, defaultBranch) {
```
Comment (oliverviljamaa) [inline lib/deployment/preparation/preparation.js:28]: Perhaps worth inlining this now, even less value in the helper function now.

File: lib/deployment/preparation/preparation.spec.js [hunk 3]
```diff
@@ -21,7 +21,28 @@ describe('Preparation', () => {
...
+    });
+
+    it('moves content from source directory to deploy directory', () => {
+      expect(shell.cp).not.toBeCalled();
+      prepareDeployDirectory('deploy-directory', options);
+      expect(shell.cp).toBeCalledWith('-r', 'source', 'deploy-directory');
+    });
+  });
+
+  describe('on a non-master branch with defaultbranch = non-master', () => {
```
Comment (oliverviljamaa) [inline lib/deployment/preparation/preparation.spec.js:44]: Hmm, what's the difference in the assertions between this and the previous `describe` block?

Comment (oliverviljamaa) [reply lib/deployment/preparation/preparation.spec.js:44]: Should we only keep the test for the changed behaviour?

Comment (andreapiras) [reply lib/deployment/preparation/preparation.spec.js]: This is checking that defaultBranch gets assigned and works as expected but can be removed

**Review by oliverviljamaa — APPROVED** [review]

**Review by oliverviljamaa — APPROVED** [review]

**Pull request merged**